fleet/comms.go

```diff
@@ -332,10 +332,24 @@ func (svc fleetCommsService) NotifyAgentGroupMemberships(a Agent) error {
 }
 
 func (svc fleetCommsService) NotifyGroupRemoval(ag AgentGroup) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
```

Comment:
Why 3 seconds on timeout? In any case, if we need this, this should be variable of the service somewhere, not hardcoded here.